### ISSUE TITLE
corrupted result with vaapi-copy for VC-1 interlaced on Polaris

### ISSUE THREAD

**Issue by aufkrawall** [mpv-player/mpv#5474]
### mpv version and platform
mpv 0.28.0-189-g7f3c7100d5 Arch Linux

### Reproduction steps
Play a VC-1 interlaced video with vaapi-copy on Polaris GPU.

### Expected behavior
Result shouldn't be corrupted.

### Actual behavior
Result shows heavy corruption:
https://abload.de/img/screenshot20180201173jxs9c.png

### Log file
https://pastebin.com/ZQtCRAW5

### Sample files
Borrowed link from Kodi:
https://drive.google.com/file/d/0BwxFVkl63-lERnhlY1pHeW5wbjQ/view
(I can't upload anything bigger by myself right now.)

**Comment by CounterPillow** [comment]
Sample rehosted on 0x0: https://0x0.st/sbla.mkv

Works fine with `vaapi-copy` on Intel, so I'd assume this is a driver bug, because you're probably the first person to have ever thrown interlaced VC-1 at an AMD hardware decoder on Linux.

**Comment by j605** [comment]
I can't reproduce it with `vdpau-copy` with `radeon` and `wayland`.

**Comment by Gusar321** [comment]
@CounterPillow Intel doesn't support interlaced VC-1 on Linux, you're getting software decoding.

**Comment by Gusar321** [comment]
I went and checked this out. libva-git and intel-vaapi-driver-git do have support for interlaced VC-1, however I too got heavy corruption with that. Then I noticed that the person who wrote interlaced VC-1 for Intel also has ffmpeg changes available: https://github.com/FFmpeg/FFmpeg/compare/master...carpalis:vc1interlace. I patched ffmpeg with that and voila, working decoding. @aufkrawall try that, see if that ffmpeg code also fixes things on Radeon. It probably will.
